### ISSUE TITLE
Rolecoins AKA Token Controlled Access

### ISSUE THREAD

**Issue by cryptonaut420** [tokenly/swapbot#17]
Need an implementation of TCA that gives the ability to define "rolecoins" for your swapbot. These rolecoins can apply various discounts and other modifiers to the machine if you hold them in the address that you are going to receive back tokens to. We dont need to hook into the Tokenly accounts system at all for this because no account is needed for purchasing stuff through swapbot, and the act of making a payment doubles as proof-of-ownership anyways, so we can just look at balances on that one address. 

Potential conditional TCA rules can get pretty complicated, but lets try and keep the interface for this as simple as possible. Enter in the token name for the rolecoin, amount required, and then what variable it modifies (either apply a modifier to a price for one of the accepted tokens, or maybe other things to such as changing the min/max purchase limits?)

**Comment by AdamBLevine** [comment]
Yes, lets start with min/max modifications and price modifications.  Rolecoin effects should be able to either effect the entire swapbot or just certain swaps per user preference, allowing different rolecoins to be used for different purposes within the same machine.

It should also be indicated whether rolecoin bonuses can stack (if two times the required rolecoin generates two times the amount, or if the only thing that matters is if it is at minimum the required amount to get the bonus, which cannot be grown) By default they should not stack.
